example/lib/frame.ex

```diff
@@ -0,0 +1,46 @@
+defmodule Frame do
+  @moduledoc """
+    Displays a list of strings
+    within a 'frame' of asterisks
+  """
+  def run(input) do
+    input
+    |> build()
+    |> print()
+  end
+
+  def build(input) do
+    longest = Enum.max_by(input, &String.length/1) |> String.length()
+
+    top = String.duplicate("*", longest + 4)
+
+    input
+    |> validate()
+    |> Enum.map(fn s ->
+      "* " <> s <> suffix(longest - String.length(s) + 1)
+    end)
+    |> List.insert_at(0, top)
+    |> List.insert_at(-1, top)
+  end
+
+  defp validate(input) do
+    case Enum.all?(input, &is_ascii?/1) do
+      true -> input
+      false -> raise ArgumentError
+    end
```

Comment:
`if else` feels nicer here (only my personal preference though)

Reply:
I'd tend to agree, but not a "blocker" to merging this PR. 
(_feel free to PR the change, happy to accept changes/simplifications_)